File: JavaScript/Splitwise/money.js
'use strict';

/** Rupees (float OK at API) → integer paise. All ledger math uses this. */
function toAmount(rupees) {
  if (typeof rupees !== 'number' || Number.isNaN(rupees)) {
    throw new Error('invalid amount');
  }
  return Math.round(rupees * 100);
}

function fromAmount(paise) {
  return paise / 100;
}

module.exports = { toAmount, fromAmount };

File: Go/Splitwise-go/money.go
package main

import (
	"fmt"
	"math"
)

// ToAmount converts rupees to integer paise. Ledger math stays in paise.
func ToAmount(rupees float64) (int64, error) {
	if math.IsNaN(rupees) || math.IsInf(rupees, 0) {
		return 0, fmt.Errorf("invalid amount")
	}
	return int64(math.Round(rupees * 100)), nil
}

func FromAmount(paise int64) float64 {
	return float64(paise) / 100
}
